src/controllers/import.js

```diff
@@ -106,25 +105,118 @@ function ImportController(context) {
     }
   }
 
+  function isMultipartFormData(headers) {
+    // Check if the Content-Type header exists and starts with "multipart/form-data"
+    const contentType = headers['Content-Type'] || headers['content-type'];
+    return contentType && contentType.startsWith('multipart/form-data');
+  }
+
+  /**
+   * Parse a multipart request to extract the URLs, options, custom import script (import.js), and
+   * custom headers.
+   * @param {Request} httpRequest - The HTTP request object.
+   * @param {object} headers - HTTP request headers object.
+   * @returns {Promise<{
+   *   urls: string,
+   *   options: object,
+   *   importScript: string,
+   *   customHeaders: object
+   * }>} Parsed request data.
+   */
+  async function parseMultipartRequest(httpRequest, headers) {
+    return new Promise((resolve, reject) => {
+      // Validate that request is multipart/form-data
+      if (!isMultipartFormData(headers)) {
+        reject(new ErrorWithStatusCode('Invalid request: expected multipart/form-data Content-Type', STATUS_BAD_REQUEST));
+      }
+
+      const busboy = Busboy({
+        headers,
+        limits: {
+          files: 1, // Limit to one file upload — import.js
+        },
+      });
+      let importScript;
+      let urls;
+      let options;
+      let customHeaders;
+
+      // Handle import.js file uploads. In absence of an import.js this callback will not be called
+      busboy.on('file', (name, file, info) => {
+        let fileBuffer = Buffer.from('');
+        const { filename, encoding, mimeType } = info;
+        log.debug(`File upload received for ${name}: filename: ${filename}, encoding: ${encoding}, mimeType: ${mimeType}`);
```

Comment:
Okay, so then it might be good to add the coralogix level logging in here to capture things like file size, who uploaded it, ims org etc.  This can be done as part of the logging story that @codevoyage is working on at a later date.